webvh/webvh/did/manager.py

```diff
@@ -0,0 +1,635 @@
+
+        server_url = self._get_server_info()
+
+        namespace = options.get("namespace")
+
+        if namespace is None:
+            raise DidCreationError("Namespace is required.")
+
+        # Generate a random identifier if not provided
+        identifier = options.get("identifier", str(uuid4()))
+
+        # Contact the server to request an identifier
+        did, challenge, domain, expiration = await self._request_identifier(
+            server_url, namespace, identifier
+        )
+
+        author_key_info = await self._get_or_create_author_key(did)
+        author_secured_document = await self._create_author_signed_document(
+            did, author_key_info, expiration, domain, challenge
+        )
+
+        result = await self._endorse_document(
+            author_secured_document, expiration, domain, challenge
+        )
+
+        if isinstance(result, dict):
+            return await self.finish_create(
+                result, state="success", author_key_info=author_key_info
+            )
+
+        try:
+            return await asyncio.wait_for(
+                self._wait_for_endorsement(did),
+                2,
+            )
+        except asyncio.TimeoutError:
+            return {
+                "status": "unknown",
+                "message": "No immediate response from endorser agent.",
+            }
+
+    async def finish_create(
+        self,
+        endorsed_document: dict,
+        state: str = "success",
+        author_key_info: Optional[dict] = None,
+    ):
+        """Finish the creation of a Webvh DID."""
+
+        if state == "posted":
+            event_bus = self.profile.inject(EventBus)
+            await event_bus.notify(
+                self.profile,
+                Event(
+                    f"endorsement_response::{endorsed_document['id']}",
+                    {"document": endorsed_document, "metadata": {"state": "posted"}},
+                ),
+            )
+            return
+
+        if state == "pending":
+            event_bus = self.profile.inject(EventBus)
+            await event_bus.notify(
+                self.profile,
+                Event(
+                    f"endorsement_response::{endorsed_document['id']}",
+                    {"document": endorsed_document, "metadata": {"state": "pending"}},
+                ),
+            )
+            return
+
+        async with ClientSession() as http_session, self.profile.session() as session:
+            # Register did document and did with the server
+            server_url = self._get_server_info()
+            response = await http_session.post(
+                server_url,
+                json={"didDocument": endorsed_document},
+                ssl=False,
+            )
+            response_json = await response.json()
+            if response.status == http.HTTPStatus.BAD_REQUEST:
+                raise DidCreationError(response_json.get("detail"))
+
+            log_entry = response_json.get("logEntry")
```

Comment:
This will change, the posting of the did document will no longer return the log entry. Instead the client must create it itself. However, I will have a helper call available by calling GET `http://example.com/my/identifier`. If this call is made after a did was registered but the log wasn't initialized, it will return a proposed initial log entry for the client to sign. However, the client should be responsible for creating this initial did log entry based on the parameters it wishes to set.